dvc/stage/__init__.py

```diff
@@ -314,7 +314,8 @@ def _changed_deps(
             if status:
                 if allow_missing and status[str(dep)] == "deleted":
                     if upstream and any(
-                        dep.fs_path == out.fs_path and dep.hash_info != out.hash_info
+                        dep.fs_path == out.fs_path
+                        and dep.hash_info.value != out.hash_info.value
```

Comment:
@pmrowla do you think it is ok?

See the added test / linked issue for the scenario, is there something different we should do for 2.X `.dvc` files that are tracked as deps in 3.X `dvc.lock` files?

Reply:
Why does `dvc status` return a clean status in this scenario?

Reply:
It does not return a clean status, it returns:

```
data_split:
        changed deps:
                deleted:            data/pool_data
...
data/pool_data.dvc:
        changed outs:
                not in cache:       data/pool_data
 ```

And it doesn't compare the `dep` of `data_split` against the `out` in `data/pool_data.dvc`

Reply:
I meant after pull, but I forgot that it's not comparing against the `out`, so ignore my comment.